osquery/registry/registry.cpp

```diff
@@ -60,14 +60,21 @@ Status RegistryHelperCore::setActive(const std::string& item_name) {
     }
   }
 
+  Status status(0, "OK");
   active_ = item_name;
   // The active plugin is setup when initialized.
   for (const auto& item : osquery::split(item_name, ",")) {
     if (exists(item, true)) {
-      Registry::get(name_, item)->setUp();
+      status = Registry::get(name_, item)->setUp();
```

Comment:
I'm a tad worried about this new status dependency but believe it's the best pattern moving forward. If a config or logger fails to setup and it is the active plugin then the core/daemon/shell should respond appropriately and exit.